its/src/test/java/com/sonar/it/shared/Tests.java

```diff
@@ -54,6 +55,16 @@ public class Tests {
     .addPlugin(MavenLocation.of("org.sonarsource.html", "sonar-html-plugin", "3.2.0.2082"))
     .build();
 
+  @ClassRule
+  public static final TemporaryFolder temp = TestUtils.createTempFolder();
+
+  @BeforeClass
+  public static void init() throws IOException {
+    TestUtils.deleteLocalCache();
+    // Analyze a project to avoid race conditions
+    analyzeProject(temp, "Empty");
```

Comment:
In theory, you can skip the msbuild part. And just run "begin" and "end", where "end" will emit some useless warning that sarif report and other stuff was not configured for C# analysis.

You would just get project dir, execute begin step, execute end step. You can make it a method in `TestUtils` that 
* would `deleteLocalCache()` - could be made private then
* would do the scanner thing

Reply:
There would also be no more accidental cache deletions